client/src/api/block_builder/input_selection/core/requirement/amount.rs

```diff
@@ -331,37 +346,13 @@ impl InputSelection {
 
             // Other kinds of outputs.
 
-            log::debug!("Trying other types of outputs");
+            let inputs = self.available_inputs.iter().filter(|input| !input.output.is_basic());
 
-            let mut inputs = self
-                .available_inputs
-                .iter()
-                .filter(|input| !input.output.is_basic())
-                .peekable();
-
-            if inputs.peek().is_some() {
-                amount_selection.fulfil(inputs);
-
-                log::debug!(
-                    "Outputs {:?} selected to fulfill the amount requirement",
-                    amount_selection.newly_selected_inputs
-                );
-                log::debug!("Triggering another amount round as non-basic outputs need to be transitioned first");
-
-                self.available_inputs
-                    .retain(|input| !amount_selection.newly_selected_inputs.contains_key(input.output_id()));
-
-                // TODO explanation of Amount
-                self.requirements.push(Requirement::Amount);
-
-                return Ok(amount_selection.into_newly_selected_inputs());
+            if self.fulfil(inputs, &mut amount_selection) {
+                break 'fulfil;
             }
         }
 
-        if amount_selection.missing_amount() != 0 {
```

Comment:
should we make this an assert or something? Just in case theres a mistake

Reply:
It's just been moved, it's still checked